internal/terraform/rootmodule/root_module.go

```diff
@@ -378,6 +378,67 @@ func (rm *rootModule) ExecuteTerraformInit(ctx context.Context) error {
 	return rm.tfExec.Init(ctx)
 }
 
+func (rm *rootModule) ExecuteTerraformValidate(ctx context.Context) (map[string]hcl.Diagnostics, error) {
```

Comment:
I reckon we'll need to find a more flexible way of integrating tfexec and the `rootmodule` functionality as adding proxy method for every command probably won't scale. Which is certainly not a criticism of your change here - This is fine in this PR as it maintains consistency, I'm just thinking out loud about the future. 😄